components/camera/videosource/webcam.go

```diff
@@ -279,19 +355,10 @@ func NewWebcam(
 	}
 	cam.Monitor()
 
-	s, err := cam.Stream(ctx)
-	if err != nil {
-		logger.Debug(err)
-	}
-
-	_, _, err = s.Next(ctx)
-	if err != nil {
-		logger.Debug(err)
-	}
-
 	return cam, nil
 }
 
+// noopCloser fulfills the projector interface during Reconfigure.
```

Comment:
Okay awesome thanks for figuring out what this is for. Note to self-  we can remove this in the eventual clean up - we dont' need a projector for pointcloud, we can wrap that logic straight into the PointCloud call using an image call.

Reply:
huh ... this seems suspect, Reconfigure should not call close  on the resource to my knowledge. Is it because VideoStream doesn't include camera resource methods (DoCommand, Close, Name and Reconfigure)

Reply:
I noticed that when reconfiguring the path, taking out the noopCloser would hang and call the real close...

Reply:
That is not how other resources reconfigure to my recollection. Either there's a netcode bug or this is behaving wildly. Always rebuild resources are the ones that follow the build -> config changes -> close -> build flow. 

Reconfiguring resources should call their reconfigure on the node in the resource graph.

Reply:
Not sure what's going on, but just re-checked

```
2024-11-13T22:41:00.280Z        INFO    rdk.resource_manager    impl/resource_manager.go:683    Now reconfiguring resource      {"resource":"rdk:component:camera/camera-webcam"}
actual close
```
```
func (c *webcam) Close(ctx context.Context) error {
+	fmt.Println("actual close")
	c.mu.Lock()
	if c.closed {
		c.mu.Unlock()
		return errors.New("webcam already closed")
	}
```

Reply:
I'm okay to leave this mystery to it's own ticket.

Reply:
Added ticket todo